eth2/simulation/src/simulation.rs

```diff
@@ -0,0 +1,88 @@
+use crate::errors::{
+    Result,
+    WhatBound,
+};
+use crate::store::Store;
+use types::eth_spec::{EthSpec};
+use types::slot_epoch_root::{
+    EEIndex,
+    Root,
+    Shard,
+    ShardSlot,
+};
+use types::execution_environment::ExecutionEnvironment;
+use types::shard_block::ShardBlock;
+use types::shard_state::ShardState;
+use types::shard_transaction::ShardTransaction;
+
+pub struct Simulation<T>
+where
+    T: EthSpec,
+{
+    store: Store<T>,
+}
+
+impl<T: EthSpec> Simulation<T> {
+    /// Add a new execution environment, return EE index
+    fn create_execution_environment(a: args::CreateExecutionEnvironment) -> EEIndex {
+        unimplemented!();
+    }
+
+    /// Add a new shard block containing a list of transactions that need to be executed
+    /// Execute all transactions on the appropriate shards / EEs, return ShardBlock index
+    fn create_shard_block(args: args::CreateShardBlock) -> ShardSlot {
+        unimplemented!();
```

Comment:
Exposing `create_shard_block()` indicates that a block proposer is not part of this node simulation.  Am I correct to assume that in order to interact with an EE deployed to this Node simulator, then I need to package one or more transactions into a block and indicate the EE each txt targets?

Reply:
Correct, though we can certainly add other interface methods in the future.  This method is intended to simulate the behavior of a block producer, so to use this endpoint you'd need to provide all the transactions, in the correct ordering, with any data required for each transaction for the given EE.